public/guides/upgrade.md

```diff
@@ -1,39 +1,241 @@
+
+Architect 7 retains full backward compatibility for existing Architect 6 projects by continuing to deploy the same API type as you're currently using.
+
+**tldr – if you have an existing Architect 6 project, you can continue to deploy that project with Architect 7. No breaking infrastructure changes will be applied by Architect 7 unless manually and explicitly opting in. However, Sandbox may be broken for your local workflows and testing until you specify which API type you're using; [see breaking changes](#architect-7-breaking-changes)**
+
+
+### Other changes
+
+- New apps will default to using `HTTP` APIs, but can be configured as `REST` APIs – more on that below
+
+
+### Removed
+
+- Removed experimental support for `@http` static mocks
+  - This was a very obscure experimental feature and should not impact anyone (but can be restored by building a macro)
+
+
+### <span id=architect-7-breaking-changes>Breaking changes
+
+All breaking changes in Architect 7 pertain to local development and testing with Sandbox, Architect's development environment. **But good news: are backward compatibility paths for Architect 6 that only require a few seconds of your time.** Read on!
+
+A core goal of Sandbox is to operate entirely locally and offline; that means no phoning home to AWS to introspect your current application configuration. Sandbox must run entirely from your machine using only your Architect project manifest. Thus, with the addition of `HTTP` APIs, Sandbox must now have a default API type.
+
+It naturally follows that Sandbox's default would be `HTTP` (with Lambda payload format version 2.0), since that is now the default for new projects. Because API Gateway `HTTP` APIs introduced breaking changes with `REST` APIs, existing projects may not function correctly until reconfigured to enter `REST` mode.
+
+
+#### Configuring the API type in Sandbox
+
+The following are the valid settings for API types:
+- **`http` (default)** - `HTTP` API + Lambda payload format version 2.0
+  - `httpv2` (aliased to `http`)
+- **`httpv1`** - `HTTP` API + Lambda payload format version 1.0; use this if you'd like to take advantage of `HTTP` APIs, but aren't yet ready to refactor your code for payload format 2.0
+- **`rest` (previous default)** - `REST` API + original API Gateway payload format (essentially the same as what is now called Lambda payload format version 1.0)
+
+You can safely toggle between `http/httpv2` and `httpv1` without destroying your current API URLs (although code may no longer function; see `@architect/functions` below for information about a seamless upgrade).
+
+Configure your project's API type to `REST` (for example) one of the following ways:
+
+1. Architect project manifest
+```arc
+# app.arc|.arc|arc.yaml|etc.
+@aws
+apigateway rest
+```
+
+or:
+
+2. Environment variable
+```sh
+ARC_API_TYPE=rest npx arc sandbox
+```
+
+
+#### Sandbox test environment changes
```

Comment:
This is an interesting section as we are describing how sandbox's API has changed. Perhaps worth considering putting the details of the sandbox API in the sandbox README, and then linking to that from here? If we do that, then I think we can eliminate the 'tl;dr' bullet point list at the end of this section and link to the sandbox readme instead. I do like the overview of the API changes near the top - that is useful info to have embedded directly into this upgrade guide.

Reply:
Excellent call. I'm also realizing I didn't update the readme for Sandbox 2.0 readme, only extensive changelog "documentation". Yeesh.

Reply:
One thing, I'm not sure I follow what you mean about the tldr bullet list in that section, I don't think it had one? I also am unclear on what overview of API changes you'd like to see at the top.

Reply:
> One thing, I'm not sure I follow what you mean about the tldr bullet list in that section, I don't think it had one? 

The last bullet point in the list on line 129, and it's sub-points too. I'll comment under it and point it out.

Reply:
Sorry, nvm! Lots of bullets and I'd already removed it in my local list.